qiskit/algorithms/optimizers/spsa.py

```diff
@@ -731,11 +731,32 @@ def _batch_evaluate(function, points, max_evals_grouped):
 
     results = []
     for batch in batched_points:
-        results += function(batch).tolist()
+        if unpack_points:
+            batch = _repack_points(batch)
+            results += _as_list(function(*batch))
+        else:
+            results += _as_list(function(batch))
 
     return results
 
 
+def _as_list(obj):
+    """Convert a list or numpy array into a list."""
+    return obj if isinstance(obj, list) else obj.tolist()
```

Comment:
It's safer to check `isinstance(obj, ndarray)` if you call `tolist` method.

Reply:
```suggestion
    return obj.tolist() if isinstance(obj, np.ndarray) else obj
```